File: 2020/05.py
def getInput():
    lines = []
    while True:
        line = input()
        if line:
            lines.append(line)
        else:
            return lines

def decode(sequence, startLower, startUpper):
    lower = startLower
    upper = startUpper
    for region in sequence:
        if region == 'F' or region == 'L':
            upper = int(upper - ((upper - lower) / 2))
        else:
            lower = int(lower + ((upper - lower) / 2)) + 1
    return lower

def decodeBoardingPass(boardingPass):
    row = decode(boardingPass[:7], 0, 127)
    column = decode(boardingPass[7:], 0, 7)
    return row, column

def getSeatId(boardingPass):
    row, column = decodeBoardingPass(boardingPass)
    return row * 8 + column

if __name__ == "__main__":
    seatIds = [getSeatId(line) for line in getInput()]
    print("max seat ID %d" % max(seatIds))

    sortedSeats = sorted(seatIds)
    previousSeat = 53
    for i in range(len(seatIds)):
        s = int(sortedSeats[i])
        if previousSeat + 1 != s:
            print("My seat ID: %d" % (previousSeat + 1))
            break
        previousSeat = s

File: 2020/05/05.c
#include <stdio.h>
#include <inttypes.h>
#include <stdlib.h>

#define BOARDING_PASS_LENGTH 10
#define INPUT_ROWS 824

struct Seating {
    uint8_t row;
    uint8_t column;
};

uint8_t decode(char *sequence, size_t sequenceSize, uint8_t startLower, uint8_t startUpper) {
    uint8_t lower = startLower;
    uint8_t upper = startUpper;
    for (size_t i = 0; i < sequenceSize; i++)
    {
        char region = sequence[i];
        if (region == 'F' || region == 'L')
            upper = upper - ((upper - lower) / 2) - 1;
        else
            lower = lower + ((upper - lower) / 2) + 1;
    }
    return lower;
}

struct Seating decodeBoardingPass(char *boardingPass) {
    struct Seating seating = {
        .row = decode(boardingPass, 7, 0, 127),
        .column = decode(boardingPass + 7, 3, 0, 7)
    };
    return seating;
}

uint16_t getSeatId(char *boardingPass) {
    struct Seating seating = decodeBoardingPass(boardingPass);
    return seating.row * 8 + seating.column;
}

int compare (const void * a, const void * b) {
  return ( *(uint16_t*)a - *(uint16_t*)b );
}

int main() {
    FILE *f = fopen ("05.txt", "r");
    char boardingPasses[INPUT_ROWS][BOARDING_PASS_LENGTH + 1];
    uint16_t allSeats[INPUT_ROWS];
    uint16_t counter = 0;
    uint16_t mySeat, highestSeatID = 0;

    while (feof(f) != 1) {
        fscanf(f, "%s", boardingPasses[counter++]);
    }
    fclose(f);

    for (size_t i = 0; i < INPUT_ROWS; i++) {
        allSeats[i] = getSeatId(boardingPasses[i]);
        if (allSeats[i] > highestSeatID)
            highestSeatID = allSeats[i];
    }

    printf("Highest Seat ID: %" PRIu16 "\n", highestSeatID);

    qsort(allSeats, sizeof(allSeats) / sizeof(uint16_t), sizeof(uint16_t), compare);

    for (size_t i = 1; i < (sizeof(allSeats) / sizeof(uint16_t)) - 1; i++) {
        if (allSeats[i - 1] == allSeats[i] - 2 && allSeats[i + 1] == allSeats[i] + 1) {
            mySeat = allSeats[i] - 1;
            break;
        }
    }

    printf("My seat ID: %" PRIu16 "\n", mySeat);
    
    return 0;
}